Lib/configparser.py

```diff
@@ -1106,10 +1106,12 @@ def _handle_rest(self, st, line, fpname):
         # a section header or option header?
         if self._allow_unnamed_section and st.cursect is None:
             st.sectname = UNNAMED_SECTION
-            st.cursect = self._dict()
-            self._sections[st.sectname] = st.cursect
-            self._proxies[st.sectname] = SectionProxy(self, st.sectname)
-            st.elements_added.add(st.sectname)
+            if st.sectname in self._sections:
+                st.cursect = self._sections[st.sectname]
+            else:
+                st.cursect = self._dict()
+                self._sections[st.sectname] = st.cursect
+                self._proxies[st.sectname] = SectionProxy(self, st.sectname)
```

Comment:
With this change, `_handle_rest` is starting to handle too much. In fact, it was probably already handling too much. Already, we can see comments (orig line 1106) that have drifted from their original purpose when this functionality came into play. I'm thinking we should extract a method for the unnamed section handling. I'll plan to do that before or after this PR.

Reply:
Found a simple way to eliminate duplicate code. Added a commit.